From 2407b82c822834e1d7408fb18ff86b11364f9fe9 Mon Sep 17 00:00:00 2001
From: Anca Jeanina FLOAREA <anca.floarea@freescale.com>
Date: Tue, 22 Oct 2013 16:04:23 +0300
Subject: [PATCH 128/130] dpa_offload: Change return error with BUG_ON

The scenario when a counter id or a request id is not properly
returned in the pool is not a normal operation and it happens
when a memory corruption occurs, case in which the calling
application is not able to recover.

Signed-off-by: Anca Jeanina FLOAREA <anca.floarea@freescale.com>
Change-Id: Id349b1a176f8fc1a17fca1c928b69bf7e35948bc
Reviewed-on: http://git.am.freescale.net:8181/5897
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Aurelian Zanoschi <Aurelian.Zanoschi@freescale.com>
Reviewed-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
[This is a patch delivered from Freescale against fsl-sdk-v1.4.5,
 rebase on current context.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_stats.c |   10 ++--------
 1 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_stats.c b/drivers/staging/fsl_dpa_offload/dpa_stats.c
index e6e2123..234c607 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_stats.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_stats.c
@@ -679,10 +679,7 @@ static int free_cnts_resources(struct dpa_stats *dpa_stats)
 		if (id != DPA_OFFLD_INVALID_OBJECT_ID) {
 			/* Release the counter id in the Counter IDs cq */
 			err = put_cnt(dpa_stats, &dpa_stats->cnts_cb[id]);
-			if (err < 0) {
-				log_err("Cannot release counter id %d\n", id);
-				return err;
-			}
+			BUG_ON(err < 0);
 		}
 		for (j = 0; j < MAX_NUM_OF_MEMBERS; j++) {
 			kfree(dpa_stats->cnts_cb[i].info.stats[j]);
@@ -798,10 +795,7 @@ static int free_reqs_resources(struct dpa_stats *dpa_stats)
 
 			/* Release the request id in the Requests IDs cq */
 			err = put_req(dpa_stats, req_cb);
-			if (err < 0) {
-				log_err("Cannot release request id %d\n", id);
-				return err;
-			}
+			BUG_ON(err < 0);
 
 			/* Release the array of counter ids */
 			kfree(req_cb->cnts_ids);
-- 
1.7.5.4

